lib/phoenix/pubsub/local.ex

```diff
@@ -0,0 +1,219 @@
+defmodule Phoenix.PubSub.Local do
+  use GenServer
+
+  @moduledoc """
+  PubSub implementation for handling local-node process groups
+
+  This modules is used by Phoenix pubsub adapters to handle their
+  local node topic subscriptions. See `Phoenix.PubSub.PG2`
+  for an example integration.
+  """
+
+  @doc """
+  Starts the server
+
+    * `server_name` - The name to registered the server under
+
+  """
+  def start_link(server_name) do
+    GenServer.start_link(__MODULE__, [], name: server_name)
+  end
+
+  @doc """
+  Subscribes the pid to the topic
+
+    * `local_server` - The registered server name or pid
+    * `pid` - The subscriber Pid
+    * `topic` - The string topic, ie "users:123"
+    * `opts` - The optional list of options. Supported options
+               only include `:link` to link the subscriber to
+               the pubsub adapter
+
+  ## Examples
+
+      iex> subscribe(:local_server, self, "foo")
+      :ok
+
+  """
+  def subscribe(local_server, pid, topic, opts \\ [])
+  def subscribe(local_server, pid, topic, link: link),
+    do: GenServer.call(local_server, {:subscribe, pid, topic, link && true})
```

Comment:
Btw, this won't be cast as a boolean if link is `nil`. Another reason for just rolling with options. :)